Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+/*global define*/
+define([
+        '../Core/BoundingSphere',
+        '../Core/Cartesian3',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/Ellipsoid',
+        '../Core/EllipsoidTangentPlane',
+        '../Core/EncodedCartesian3',
+        '../Core/IndexDatatype',
+        '../Core/Math',
+        '../Core/Matrix4',
+        '../Core/PolygonGeometryLibrary',
+        '../Core/PolygonPipeline',
+        '../Core/PrimitiveType',
+        '../Core/WindingOrder',
+        '../Renderer/BufferUsage',
+        '../Renderer/DrawCommand',
+        '../Renderer/ShaderSource',
+        '../Shaders/ShadowVolumeFS',
+        '../Shaders/ShadowVolumeVS',
+        './BlendingState',
+        './CullFace',
+        './Pass',
+        './StencilFunction',
+        './StencilOperation'
+    ], function(
+        BoundingSphere,
+        Cartesian3,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        Ellipsoid,
+        EllipsoidTangentPlane,
+        EncodedCartesian3,
+        IndexDatatype,
+        CesiumMath,
+        Matrix4,
+        PolygonGeometryLibrary,
+        PolygonPipeline,
+        PrimitiveType,
+        WindingOrder,
+        BufferUsage,
+        DrawCommand,
+        ShaderSource,
+        ShadowVolumeFS,
+        ShadowVolumeVS,
+        BlendingState,
+        CullFace,
+        Pass,
+        StencilFunction,
+        StencilOperation) {
+    "use strict";
+
+    var GroundPolygon = function(options) {
```

Comment:
This is going to become a geometry that is provided to `Primitive`, right?  That is what you mean by "Hook up batching?"